runtime/rastrace/method_trace.c

```diff
@@ -172,45 +172,82 @@ traceMethodEnter(J9VMThread *thr, J9Method *method, void *receiverAddress, UDATA
 	J9UTF8* methodName = J9ROMMETHOD_NAME(J9_ROM_METHOD_FROM_RAM_METHOD(method));
 	J9UTF8* methodSignature = J9ROMMETHOD_SIGNATURE(J9_ROM_METHOD_FROM_RAM_METHOD(method));
 	UDATA modifiers = J9_ROM_METHOD_FROM_RAM_METHOD(method)->modifiers;
+	j9object_t receiver = J9_ARE_ANY_BITS_SET(modifiers, J9AccStatic) ? NULL : *(j9object_t *)receiverAddress;
 
-	if (isCompiled) {
-		if (modifiers & J9AccStatic) {
+	if (0 != isCompiled) {
+		if (J9_ARE_ANY_BITS_SET(modifiers, J9AccStatic)) {
 			Trc_MethodEntryCS(thr, J9UTF8_LENGTH(className), J9UTF8_DATA(className), J9UTF8_LENGTH(methodName), J9UTF8_DATA(methodName), J9UTF8_LENGTH(methodSignature), J9UTF8_DATA(methodSignature));
 		} else {
-			j9object_t receiver = *(j9object_t*)receiverAddress;
 			Trc_MethodEntryC(thr, J9UTF8_LENGTH(className), J9UTF8_DATA(className), J9UTF8_LENGTH(methodName), J9UTF8_DATA(methodName), J9UTF8_LENGTH(methodSignature), J9UTF8_DATA(methodSignature), receiver);
 		}
 	} else {
 		char buf[1024];
 
-		if (modifiers & J9AccStatic) {
-			if (modifiers & J9AccNative) {
+		if (J9_ARE_ANY_BITS_SET(modifiers, J9AccStatic)) {
+			if (J9_ARE_ANY_BITS_SET(modifiers, J9AccNative)) {
 				Trc_MethodEntryNS(thr, J9UTF8_LENGTH(className), J9UTF8_DATA(className), J9UTF8_LENGTH(methodName), J9UTF8_DATA(methodName), J9UTF8_LENGTH(methodSignature), J9UTF8_DATA(methodSignature));
 			} else {
 				Trc_MethodEntryS(thr, J9UTF8_LENGTH(className), J9UTF8_DATA(className), J9UTF8_LENGTH(methodName), J9UTF8_DATA(methodName), J9UTF8_LENGTH(methodSignature), J9UTF8_DATA(methodSignature));
 			}
-			if (doParameters) {
+			if (0 != doParameters) {
 				Trc_MethodArgumentsS(thr, traceMethodArguments(thr, methodSignature, (UDATA*)receiverAddress, buf, buf + sizeof(buf)));
 			}
 		} else {
-			j9object_t receiver = *(j9object_t*)receiverAddress;
-			J9Class* receiverClazz = J9OBJECT_CLAZZ(thr, receiver);
-			J9UTF8* receiverClassName = J9ROMCLASS_CLASSNAME(receiverClazz->romClass);
-
-			if (modifiers & J9AccNative) {
+			if (J9_ARE_ANY_BITS_SET(modifiers, J9AccNative)) {
 				Trc_MethodEntryN(thr, J9UTF8_LENGTH(className), J9UTF8_DATA(className), J9UTF8_LENGTH(methodName), J9UTF8_DATA(methodName), J9UTF8_LENGTH(methodSignature), J9UTF8_DATA(methodSignature), receiver);
 			} else {
 				Trc_MethodEntry(thr, J9UTF8_LENGTH(className), J9UTF8_DATA(className), J9UTF8_LENGTH(methodName), J9UTF8_DATA(methodName), J9UTF8_LENGTH(methodSignature), J9UTF8_DATA(methodSignature), receiver);
 			}
+		}
+		if (doParameters) {
```

Comment:
The test must be like line 192:
```c
		if (0 != doParameters) {
```

This must be structured so we don't get parameters from line 193 in addition to the code below.